ts/utils/fetch.ts

```diff
@@ -56,6 +64,28 @@ export function defaultRetryingFetch(
   return retryingFetch((global as any).fetch, timeout, maxRetries);
 }
 
+function retryLogicForTransientResponseError(
```

Comment:
add a comment describing the function + I would like to have some tests for this function

Reply:
d42a685 implements test case